docs/modelserving/v1beta1/amd/README.md

```diff
@@ -0,0 +1,189 @@
+    datatype: "FP32"
+    shape: [28,28,1]
+  }
+]
+outputs [
+  {
+    name: "flatten/Reshape"
+    datatype: "FP32"
+    shape: [10]
+  }
+]
+```
+
+The name, platform, inputs and outputs must all be defined.
+Some notes about the acceptable values:
+- the name must uniquely identify a model for the server
+- the platform must be one of `tfzendnn_graphdef` or `vitis_xmodel`
+- for each input/output, the name, datatype, and shape must be defined
+  - the name corresponds to the name of the input/output tensor(s). Multiple input/output tensors aren't currently supported
+  - the shape must be a flat array describing the shape of the input/output tensor
+
+The final model directory structure looks like:
+
+```
+/
+├─ model_a/
+│  ├─ 1/
+│  │  ├─ saved_model.pb
+│  ├─ config.pbtxt
+```
+
+The names for the files (`saved_model.x` and `config.pbtxt`) must match as above.
+The file extension for `tfzendnn_graphdef` and `vitis_xmodel` models should be `.pb` and `.xmodel`, respectively.
+This model can now be zipped and stored on a web server or made available on a cloud storage platform compatible with KServe.
+
+## Make an inference
+
+The AMD Inference Server can be used in single or multi-model serving mode in KServe.
+The code snippets below use the environment variables `INGRESS_HOST` and `INGRESS_PORT` to make requests to the cluster.
+[Find the ingress host and port](https://kserve.github.io/website/master/get_started/first_isvc/#4-determine-the-ingress-ip-and-ports) for making requests to your cluster and set these values appropriately.
+
+### Single model serving
+
+```bash
+# download the inference service file and input data
+curl -O https://raw.githubusercontent.com/kserve/kserve/master/docs/samples/v1beta1/amd/single_model.yaml
+curl -O https://raw.githubusercontent.com/kserve/kserve/master/docs/samples/v1beta1/amd/input.json
+
+# update the single_model.yaml to use your custom image
+sed -i "s/<image>/$(whoami)\/proteus:latest/" single_model.yaml
+
+# create the inference service
+kubectl apply -f single_model.yaml
+
+# wait for service to be ready
+kubectl wait --for=condition=ready isvc -l app=example-amdserver-single-isvc
+
+export SERVICE_HOSTNAME=$(kubectl get inferenceservice example-amdserver-single-isvc -o jsonpath='{.status.url}' | cut -d "/" -f 3)
+```
+
+### Multi model serving
+
+```bash
+# download the inference service file and input data
+curl -O https://raw.githubusercontent.com/kserve/kserve/master/docs/samples/v1beta1/amd/multi_model.yaml
+curl -O https://raw.githubusercontent.com/kserve/kserve/master/docs/samples/v1beta1/amd/input.json
+
+# update the multi_model.yaml to use the custom image
+sed -i "s/<image>/$(whoami)\/proteus:latest/" multi_model.yaml
+
+# create the inference service
+kubectl apply -f multi_model.yaml
+
+# wait for service to be ready
+kubectl wait --for=condition=ready isvc -l app=example-amdserver-multi-isvc
+
+# add a model to serve: our MNIST model
+kubectl apply -f trained_model.yaml
```

Comment:
We are going to deprecate trained model soon, alternatively you can create a inference service with `StorageUri` pointing to multiple models if you want to demonstrate the multi model capability. Or you can integrate AMD runtime with ModelMesh as you are now able to support v2 protocol.

Reply:
I see, what's the timeline for the deprecation? ModelMesh should be the long-term solution but that will take some time. As for the `StorageUri` option, doesn't that only work with the `ServingRuntimes` approach, but not the custom predictor I'm using in `single_model.yaml`? There, the storage URI is specified as an environment variable.